certifier/src/certifier.rs

```diff
@@ -20,71 +23,240 @@ pub struct CertifyRequest {
 }
 
 #[serde_as]
-#[derive(Debug, Serialize)]
-struct CertifyResponse {
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CertifyResponse {
+    /// The certificate as scale-encoded `Certificate` struct
     #[serde_as(as = "Base64")]
-    signature: Vec<u8>,
+    pub certificate: Vec<u8>,
+    /// Signature of the certificate
     #[serde_as(as = "Base64")]
-    pub_key: Vec<u8>,
+    pub signature: Vec<u8>,
+    /// The public key of the certifier that signed the certificate
+    #[serde_as(as = "Base64")]
+    pub pub_key: Vec<u8>,
+}
+
+#[derive(Debug, Decode, Encode)]
+pub struct Certificate {
+    // ID of the node being certified
+    pub pub_key: Vec<u8>,
+    /// Unix timestamp
+    pub expiration: Option<Compact<u64>>,
 }
```

Comment:
Maybe a bit late, but we could have gone the route of using actual x509 certificates and then use client authentication (mTLS) against the certifier, would have probably been less implementation effort and made it easier to develop 3rd party tools or alternative implementations.

Reply:
Could a x509 certificate be only valid for a single nodeID? A certificate should allow for only 1 poet registration per round.

Reply:
Yes, a certificate would be for a specific public key until a given expiration date. I don't think the effort to change it now is worth it though.